ksp_plugin_adapter/main_window.cs

```diff
@@ -0,0 +1,493 @@
+            options : UnityEngine.GUILayout.Width(column_width));
+        UnityEngine.GUILayout.Toggle(
+            value   : severity >= Log.GetSuppressedLogging(),
+            text    : "",
+            options : UnityEngine.GUILayout.Width(column_width));
+        UnityEngine.GUILayout.Toggle(
+            value   : severity >= Log.GetStderrLogging(),
+            text    : "",
+            options : UnityEngine.GUILayout.Width(column_width));
+        UnityEngine.GUILayout.Toggle(
+            value   : severity > Log.GetBufferedLogging(),
+            text    : "",
+            options : UnityEngine.GUILayout.Width(column_width));
+      }
+    }
+    using (new UnityEngine.GUILayout.HorizontalScope()) {
+      UnityEngine.GUILayout.Space(column_width);
+      if (UnityEngine.GUILayout.Button(
+              text    : "↓",
+              options : UnityEngine.GUILayout.Width(column_width))) {
+        Log.SetSuppressedLogging(Math.Min(suppressed_logging_ + 1, 3));
+        suppressed_logging_ = Log.GetSuppressedLogging();
+      }
+      if (UnityEngine.GUILayout.Button(
+              text    : "↓",
+              options : UnityEngine.GUILayout.Width(column_width))) {
+        Log.SetStderrLogging(Math.Min(stderr_logging_ + 1, 3));
+        stderr_logging_ = Log.GetStderrLogging();
+      }
+      if (UnityEngine.GUILayout.Button(
+              text    : "↓",
+              options : UnityEngine.GUILayout.Width(column_width))) {
+        Log.SetBufferedLogging(Math.Min(buffered_logging_ + 1, 3));
+        buffered_logging_ = Log.GetBufferedLogging();
+      }
+    }
+    UnityEngine.GUILayout.TextArea("Journalling is " +
+                                   (journaling_ ? "ON" : "OFF"));
+    must_record_journal_ = UnityEngine.GUILayout.Toggle(
+        value   : must_record_journal_,
+        text    : "Record journal (starts on load)");
+    if (journaling_ && !must_record_journal_) {
+      // We can deactivate a recorder at any time, but in order for replaying to
+      // work, we should only activate one before creating a plugin.
+      journaling_ = false;
+      Interface.ActivateRecorder(false);
+    }
+  }
+
+  private void RenderPredictionSettings() {
+    if (vessel_ != predicted_vessel_()) {
+      vessel_ = predicted_vessel_();
+      string vessel_guid = vessel_?.id.ToString();
+      if (vessel_guid != null && plugin_.HasVessel(vessel_guid)) {
+        AdaptiveStepParameters adaptive_step_parameters =
+            plugin_.VesselGetPredictionAdaptiveStepParameters(vessel_guid);
+        prediction_length_tolerance_index_ = Array.FindIndex(
+            prediction_length_tolerances_,
+            (double tolerance) =>
+                tolerance >=
+                    adaptive_step_parameters.length_integration_tolerance);
+        prediction_steps_index_ = Array.FindIndex(
+            prediction_steps_,
+            (Int64 step) => step >= adaptive_step_parameters.max_steps);
```

Comment:
What happens if `Array.FindIndex` returns `-1`?

Reply:
Added handling of that case.